controller/controller.go

```diff
@@ -390,14 +390,18 @@ func ScaleUp(
 		"scaleUpLimit": scaleUpLimit,
 	})
 
-	configYAML, _ := models.NewConfigYAML(scheduler.YAML)
+	configYAML, err := models.NewConfigYAML(scheduler.YAML)
+	if err != nil {
+		return fmt.Errorf("failed to load scheduler configuration: %w", err)
+	}
 
-	existPendingPods, err := pendingOrFailedPods(config, redisClient, scheduler.Name, mr)
+	hasNotReadyPods, err := checkNotReadyPods(config, redisClient, scheduler.Name, configYAML.PreventRoomsCreationWithError, mr)
```

Comment:
minor: When you say: check not ready pods, I read: "if true, everything is fine, else, there're not ready pods"

WDYT of change the name to hasNotReadyPods? In this way I believe things get more clear.